cloud/scope/cluster_test.go

```diff
@@ -713,3 +713,204 @@ func TestDeleteSubnet(t *testing.T) {
 		})
 	})
 }
+
+func TestCreateLoadBalancer(t *testing.T) {
+	var (
+		mockvpc  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockvpc = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	vpcCluster := infrav1beta1.IBMVPCCluster{
+		Spec: infrav1beta1.IBMVPCClusterSpec{
+			ControlPlaneLoadBalancer: &infrav1beta1.VPCLoadBalancerSpec{
+				Name: "foo-load-balancer",
+			},
+		},
+		Status: infrav1beta1.IBMVPCClusterStatus{
+			Subnet: infrav1beta1.Subnet{
+				ID: core.StringPtr("foo-subnet-id"),
+			},
+		}}
+	expectedOutput := &vpcv1.LoadBalancer{
+		Name: core.StringPtr("foo-load-balancer"),
+	}
+
+	t.Run("Create LoadBalancer", func(t *testing.T) {
+		listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
+		createLoadBalancerOptions := &vpcv1.CreateLoadBalancerOptions{}
```

Comment:
good to avoid these global pointer variables, lets create them while calling those respective apis, e.g:

```golang
			mockvpc.EXPECT().ListLoadBalancers(gomock.AssignableToTypeOf(&vpcv1.ListLoadBalancersOptions{})).Return(loadBalancerCollection, &core.DetailedResponse{}, nil)
```

Reply:
Lets fix this everywhere.

Reply:
Fixed.